src/theory/builtin/theory_builtin_rewriter.h

```diff
@@ -62,31 +62,58 @@ class TheoryBuiltinRewriter {
   /** recursive helper for getArrayRepresentationForLambda */
   static Node getArrayRepresentationForLambdaRec( TNode n, bool reqConst, TypeNode retType );
 public:
-  /** 
-   * Given an array constant a, returns a lambda expression that it corresponds to, with bound variable list bvl. 
-   * Examples:
-   *
-   * (store (storeall (Array Int Int) 2) 0 1) 
-   * becomes
-   * ((lambda x. (ite (= x 0) 1 2))
-   *
-   * (store (storeall (Array Int (Array Int Int)) (storeall (Array Int Int) 4)) 0 (store (storeall (Array Int Int) 3) 1 2))
-   * becomes
-   * (lambda xy. (ite (= x 0) (ite (= x 1) 2 3) 4))
-   *
-   * (store (store (storeall (Array Int Bool) false) 2 true) 1 true)
-   * becomes
-   * (lambda x. (ite (= x 1) true (ite (= x 2) true false)))
-   *
-   * Notice that the return body of the lambda is rewritten to ensure that the representation is canonical. Hence the last
-   * example will in fact be returned as:
-   * (lambda x. (ite (= x 1) true (= x 2)))
+ /** Get function type for array type
+ * This returns the function type of terms returned by the function
+ * getLambdaForArrayRepresentation( t, bvl ),
+ * where t.getType()=atn.
+ *
+ * bvl should be a bound variable list whose variables correspond to in-order to
+ * the index types of the
+ * (curried) Array type. For example a bound variable list bvl whose variables
+ * have types (Int, Real) would be a proper
+ * input when paired with atn = (Array Int (Array Real Bool)), or (Array Int
+ * (Array Real (Array Bool Bool))).
+ * This function returns (-> Int Real Bool) and (-> Int Real (Array Bool Bool))
+ * respectively in these cases.
+ * Whereas bvl is not a proper input for atn = (Array Int (Array Bool Bool)) or
+ * (Array Int Int).
+ */
```

Comment:
Fix formatting (hard to read).

(I know, this is mainly due to the fact that clang-format simply breaks comment lines that exceed the 80 char limit, but as of yet there's no way to let clang-format handle that in a better way. Unfortunately this means that we will sometimes have to fix this manually.)